osu.Game/OsuGame.cs

```diff
@@ -987,7 +983,7 @@ protected virtual void ScreenChanged(IScreen current, IScreen newScreen)
 
                 if (newOsuScreen.HideOverlaysOnEnter)
                     CloseAllOverlays();
-                else
+                else if (!Toolbar.HiddenByUser)
                     Toolbar.Show();
```

Comment:
This sounds like it should be done inside the toolbar component (see `UpdateState(...)`), instead of exposing "hidden by user" flag publicly and checking before attempting to show in each usage